### PR TITLE
First Draft: Cookies with Nextjs

### LINKED ISSUE #29599: Guide for how to set or read cookies
### Describe the feature you'd like to request

We don't currently have a dedicated guide for how to set or read cookies. This would be valuable information to our users.

### Describe the solution you'd like

Create a new guide which outlines the how, when and why or setting and reading cookies.

### Describe alternatives you've considered

N/A

### PR DESCRIPTION
**Pull request by MaedahBatool**
We don't currently have a dedicated guide for how to set or get cookies with Next.js. This would be valuable information to our users.

## Feature

- [x] Related issues linked using  #29599
- [x] Documentation added

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: docs/guides/next-cookies.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+You can read more details about the `document.cookie` property on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie).
+
+### Set and Get Cookies on the Server-side on the Client-side
+
+Let's build a demo application to set a `theme` cookie to save the `light` or `dark` mode of your website.
+
+![client-side-cookies](/docs/guides/images/client-cookies.png)
+
+Let's create a Next.js page that shows the current value of the cookie`theme` and allows you to change it to `light` or `dark`.
```
Comment (molebox) [inline docs/guides/next-cookies.md:47]:
```suggestion
First create a Next.js page that shows the current value of the cookie`theme` and allows you to change it to `light` or `dark`.
```

File: docs/guides/next-cookies.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+```js
```
Comment (molebox) [inline docs/guides/next-cookies.md:49]: This codeblock has the `changeTheme` function, without an explanation of what it is. Maybe we could just use the second codeblock with the full code instead?

File: docs/guides/next-cookies.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+You can use two React.js hooks to make this app easier to work with:
+
+- [`useState`](https://reactjs.org/docs/hooks-state.html) hook to save the `theme` cookie value
+- [`useEffect`](https://reactjs.org/docs/hooks-effect.html) hook to access the `document`, which is a side-effect
+
+To understand how React hooks work, [check out the React documentation](https://reactjs.org/docs/hooks-intro.html).
+
+Here's the basic structure of our code so far:
+
+```js
```
Comment (molebox) [inline docs/guides/next-cookies.md:71]: Same as above, i think we can merge these codeblocks with the final code

File: docs/guides/next-cookies.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+}
+```
+
+This code sets `light` mode as the default theme. Then it allows the user to click a button to change that to `dark` mode or back to `light` mode. The state is kept by the `useState` React hook, which updates the UI when the cookie value is changed.
+
+### TODO: _ADD VIDEO SET CLIENT COOKIES_
+
+## Server-side cookies with Next.js
+
+Working with cookies on the server-side with Next.js is just like you would expect. However, Next.js provides you with a `context.req.cookies` JavaScript object containing all the cookies.
```
Comment (molebox) [inline docs/guides/next-cookies.md:145]:
```suggestion
Next.js provides you with a `context.req.cookies` JavaScript object containing all the cookies.
```

File: docs/guides/next-cookies.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+If you export an async function called [`getServerSideProps`](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) from a page, Next.js will pre-render this page on each request using the data returned by `getServerSideProps`.
```
Comment (molebox) [inline docs/guides/next-cookies.md:147]:
```suggestion
If you export an async function called [`getServerSideProps`](/docs/basic-features/data-fetching/get-server-side-props) from a page, Next.js will pre-render this page on each request using the data returned by `getServerSideProps`.
```

**Review by molebox — COMMENTED** [review]
I think it would be cool to add an example using middleware too. Also, will this conflict with your forms guides? As the guides section doesnt exist yet. No biggie, easy fix

File: docs/guides/next-cookies.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+```js
+import { useEffect, useState } from 'react'
+
+export default function ClientSideCookiePage() {
+  const [theme, setTheme] = useState()
+
+  useEffect(() => {
+    // Get cookies & find the value of the cookie called `theme`.
+    const color = document.cookie
```
Comment (neeraj3029) [inline docs/guides/next-cookies.md:107]: docs example must use `decodeURIComponent` on the cookie value before returning.

File: docs/guides/next-cookies.md [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+      ?.split(';')
+      .find((x) => x.trim().startsWith('theme'))
+      ?.split('=')[1]
+
+    // Set default cookie to `light` if not already set.
+    if (!color) {
+      color = 'light'
+      document.cookie = `theme=${color}`
```
Comment (neeraj3029) [inline docs/guides/next-cookies.md:115]: Similarly, we must call `encodeURIComponent` on cookie value before setting.

**Comment by leerob** [comment]
Going to close this out for now since it's been paused 🙏

**Pull request closed without merging**